src/pages/home/report/ReportActionsList.js

```diff
@@ -87,34 +94,116 @@ function keyExtractor(item) {
     return item.reportActionID;
 }
 
+function isUnreadMsg(message, lastRead) {
+    return message && lastRead && message.created && lastRead < message.created;
+}
+
 function ReportActionsList(props) {
     const reportScrollManager = useReportScrollManager();
     const opacity = useSharedValue(0);
+    const userActiveSince = useRef(null);
+    const currentUnreadMarker = useRef(null);
+    const scrollingVerticalOffset = useRef(0);
+    const readActionSkipped = useRef(false);
+    const [messageManuallyMarked, setMessageManuallyMarked] = useState(false);
+    const report = props.report;
+    const sortedReportActions = props.sortedReportActions;
+    const [isFloatingMessageCounterVisible, setIsFloatingMessageCounterVisible] = useState(false);
+    const [reportActionSize, setReportActionSize] = useState(sortedReportActions.length);
```

Comment:
We need to fire a rerender, there was a race condition the unread marker wasn't getting re-calculated because the rerender produced by onyx (when reports get updated) ran before the clearing `currentUnreadMarker.current = null;` so we couldn't use the onyx rerender for recalculations

Reply:
Just gave it around of testing using ref and is working, so I will remove the `useState(sortedReportActions.length);`